tuiexporter/internal/tui/component/page/metric/chart_test.go

```diff
@@ -89,51 +92,85 @@ func TestChartInputCaptureAfterFlush(t *testing.T) {
 	assert.Nil(t, got)
 }
 
-func TestGetDataToDrawWithMoreKeysThanColors(t *testing.T) {
-	// Create dataMap with 12 keys (more than 10 colors available)
-	// This test ensures no panic occurs when there are more keys than colors
-	attrkey := "test_attr"
-	numKeys := 12
-
-	dataMap := make(map[string]map[string][]*pmetric.NumberDataPoint)
-	dataMap[attrkey] = make(map[string][]*pmetric.NumberDataPoint)
-
-	start := time.Now()
-	end := start.Add(time.Minute)
-
-	for i := 0; i < numKeys; i++ {
-		key := fmt.Sprintf("key_%02d", i)
-		// Use multiple data points per key to exercise interpolation logic
-		dps := make([]*pmetric.NumberDataPoint, 0, 3)
-		for j := 0; j < 3; j++ {
-			dp := pmetric.NewNumberDataPoint()
-			// Alternate between double and int values
-			if (i+j)%2 == 0 {
-				dp.SetDoubleValue(float64(i*10 + j))
-			} else {
-				dp.SetIntValue(int64(i*10 + j))
-			}
-			dp.SetTimestamp(pcommon.NewTimestampFromTime(start.Add(time.Duration(i*3+j) * time.Second)))
-			dps = append(dps, &dp)
-		}
-		dataMap[attrkey][key] = dps
+func TestLineColors(t *testing.T) {
+	c := layout.Colors
+	n := len(c)
+
+	tests := []struct {
+		name    string
+		numKeys int
+		want    []tcell.Color
+	}{
+		{
+			name:    "within bounds",
+			numKeys: n - 1,
+			want:    c,
+		},
+		{
+			name:    "exact bound",
+			numKeys: n,
+			want:    c,
+		},
+		{
+			name:    "one over",
+			numKeys: n + 1,
+			want:    append(append([]tcell.Color{}, c...), c[0]),
+		},
 	}
 
-	chart := newChart(layout.NewCommandList(), nil, []*layout.ResizeManager{})
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			attrkey := "test_attr"
+			dataMap := make(map[string]map[string][]*pmetric.NumberDataPoint)
+			dataMap[attrkey] = make(map[string][]*pmetric.NumberDataPoint)
+
+			start := time.Now()
+			end := start.Add(time.Minute)
+
+			for i := 0; i < tt.numKeys; i++ {
+				dp := pmetric.NewNumberDataPoint()
+				dp.SetDoubleValue(float64(i))
+				dp.SetTimestamp(pcommon.NewTimestampFromTime(start.Add(time.Duration(i) * time.Second)))
+				dataMap[attrkey][fmt.Sprintf("key_%02d", i)] = []*pmetric.NumberDataPoint{&dp}
+			}
 
-	// This should not panic
-	data, tv := chart.getDataToDraw(dataMap, attrkey, start, end)
+			chart := newChart(layout.NewCommandList(), nil, []*layout.ResizeManager{})
 
-	// Verify we got data for all 12 keys
-	assert.Len(t, data, numKeys)
+			data, _ := chart.getDataToDraw(dataMap, attrkey, start, end)
 
-	// Verify the legend text contains all keys with colors (colors wrap around)
-	legendText := tv.GetText(false)
-	for i := 0; i < numKeys; i++ {
-		key := fmt.Sprintf("key_%02d", i)
-		assert.Contains(t, legendText, key)
-		// Verify color tag is present (colors wrap via modulo)
-		expectedColor := layout.Colors[i%len(layout.Colors)].String()
-		assert.Contains(t, legendText, expectedColor)
+			assert.Equal(t, tt.want, lineColors(len(data)))
+		})
 	}
 }
+
+func TestDrawMetricNumberChartWithManyDataPoints(t *testing.T) {
```

Comment:
this is the real test that catches both panics (one from before and also the missing amount of colors)